services/cache/dns.c

```diff
@@ -654,6 +655,10 @@ tomsg(struct module_env* env, struct query_info* q, struct reply_info* r,
 	msg->rep->rrset_count = r->rrset_count;
 	msg->rep->authoritative = r->authoritative;
 	msg->rep->reason_bogus = r->reason_bogus;
+	if(r->reason_bogus_str) {
+		msg->rep->reason_bogus_str = regional_strdup(region, r->reason_bogus_str);
```

Comment:
The variable `msg->rep->reason_bogus_str` was not zeroed by `gen_dns_msg`, so it should be set to NULL when `r->reason_bogus_str` is NULL, I think?

Reply:
It is set to NULL in `gen_dns_msg`.

Reply:
That is correct. I was looking in the master branch, for that.